diffrax/_solver/srk.py

```diff
@@ -402,41 +525,53 @@ def aux_add_levy(w_leaf, *levy_leaves):
         # where levy is either H or K (if those entries exist)
         # this is similar to h_kfs or w_kgs, but for the Levy area(s)
 
-        if additive_noise:
+        if isinstance(cfs_bm, AdditiveBrownianCoeffs):  # additive noise
             # compute g once since it is constant
-            g0 = diffusion.vf(t0, y0, args)
-            w_kg = diffusion.prod(g0, w)
-            a_w = jnp.asarray(self.tableau.cfs_w.a, dtype=dtype)
-            if stla:
+            def _comp_g(_, _t):
+                return None, diffusion.vf(_t, y0, args)
+
+            _, (g0, g1) = lax.scan(_comp_g, None, jnp.array([t0, t1]), length=2)
```

Comment:
I think there might be an error here (or maybe just a change in interface). Previous code based on this branch was working that now breaks and I believe this change might be the cause. 

My current thesis is that the indexing of g0, g1 does not act as intended. I believe what this is supposed to do is get g0 (for t0) and g1 (for t1). However, indexing like this does not support all pytrees. E.g. if the diffusion vf returns a pytree, say a dictionary, (g0, g1) will get the keys but not the values. Or if it were an equinox module, indexing of module[0] doesn't get all the first values of the arrays. I believe this is equivalent to `g0, g1 = diffusion.vf(t0, y0, args), diffusion.vf(t1, y0, args)` and when I replaced this locally things seemed to work. Not sure about relative speed or confident on correctness, just something I tried.

If the above is correct, it might be good to add a test for it. If I am simply using it wrong, please let me know. I can also provide the MVC if needed.

Reply:
Thanks for letting ne know! The reason for this change is the following comment:
https://github.com/patrick-kidger/diffrax/pull/344#discussion_r1492909600

But I see where I went wrong, I'll fix it later today.

I agree, we should add a test_sde_shape_dtype somewhere. Could you suggest some code for it? Maybe something along the lines of `test_basic` from `test_integrate.py`.

Reply:
You don't seem to have any sequential dependence here -- could just do this as a `jax.vmap`?

Reply:
(+1 for adding a test to this though, because this kind of "unpacking of pytrees" error is a common class of error that I see)